client/api/src/client.rs

```diff
@@ -63,6 +67,9 @@ pub trait BlockchainEvents<Block: BlockT> {
 	/// finalized block.
 	fn finality_notification_stream(&self) -> FinalityNotifications<Block>;
 
+	/// Get a stream of finality notification originating from the initial block import
```

Comment:
It's not `finality`, it's simply block import. I think we should also avoid `initial_sync` in naming. It should be `all_blocks_import_notification_stream` or sth like this and we should emphasise that this is triggered on every imported block, not like in case of regular `import_notification_stream`.